go/vt/discovery/healthcheck_test.go

```diff
@@ -63,6 +63,77 @@ func init() {
 	refreshInterval = time.Minute
 }
 
+func TestNewVTGateHealthCheckFilters(t *testing.T) {
+	defer func() {
+		KeyspacesToWatch = nil
+		tabletFilters = nil
+		tabletFilterTags = nil
+	}()
+
+	testCases := []struct {
+		name                string
+		keyspacesToWatch    []string
+		tabletFilters       []string
+		tabletFilterTags    map[string]string
+		expectedError       string
+		expectedFilterTypes []any
+	}{
+		{
+			name: "noFilters",
+		},
+		{
+			name:                "tabletFilters",
+			tabletFilters:       []string{"ks1|-80"},
+			expectedFilterTypes: []any{&FilterByShard{}},
+		},
+		{
+			name:                "keyspacesToWatch",
+			keyspacesToWatch:    []string{"ks1"},
+			expectedFilterTypes: []any{&FilterByKeyspace{}},
```

Comment:
Same thing here, and for the others. It's not obvious to me how we're testing the specific behavior (I do see that we test that we get the expected *type* of filter(s). But don't we want to test that 1) we get the specific filters we expect and 2) the filtering works as expected (meaning that specific tablets are ignored or filtered out?

Reply:
@mattlord those tests are in `topology_watcher_test.go`